Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx

```diff
@@ -0,0 +1,234 @@
+                {this.renderInterval()}
+                {this.renderWindow()}
+                {this.renderTimeRange()}
+            </div>
+        );
+    }
+
+    renderAlias() {
+        return (
+            <div style={sectionStyle}>
+                <TextField
+                    className="ms-font-m info-settings-textbox"
+                    spellCheck={false}
+                    label="Alias"
+                    value={this.props.batch.id}
+                    onChange={(event, value) => this.props.onUpdateBatchName(value)}
+                    onGetErrorMessage={value => this.validateProperty(value)}
+                />
+            </div>
+        );
+    }
+    
+    renderTypeDisplayName() {
+        return (
+            <div style={batchTypeSection}>
+                <TextField
+                    className="ms-font-m info-settings-textbox"
+                    spellCheck={false}
+                    label="Batch Type"
+                    disabled={true}
+                    value={this.props.batchTypeDisplayName}
+                />
+            </div>
+        );
+    }
+
+    renderTimeRange() {
+        return (
+            <div >
+                {this.renderStartTime()}
+                {this.renderEndTime()}
+            </div>
+        );
+    }
+    
+    renderStartTime() {
+        return (
+            <div style={sectionStyle}>
+                <Label className="ms-font-m info-settings-textbox">Start Time</Label>
+                <DateTimePicker
+                    className="ms-font-m info-settings-textbox"
+                    value={this.props.batch.properties.startTime}
+                    onChange={(value) => this.props.onUpdateBatchStartTime(value)}
+                />
+            </div>
+        );
+    }
+
+    renderEndTime() {
+        return (
+            <div style={sectionStyle}>
+                <Label className="ms-font-m info-settings-textbox">End Time</Label>
+                <DateTimePicker
+                    className="ms-font-m info-settings-textbox"
+                    value={this.props.batch.properties.endTime}
+                    onChange={(value) => this.props.onUpdateBatchEndTime(value)}
+                />
+            </div>
+        );
+    }
+
+    renderInterval() {
+        return (
+            <div style={sectionContainerStyle}>
+                {this.renderIntervalText()}
+                {this.renderIntervalTypeDropdown()}
+            </div>
+        );
+    }
+
+    renderIntervalText() {
+        return (
+            <div style={sectionValueStyle}>
+                <TextField
+                    className="ms-font-m"
+                    spellCheck={false}
+                    label="Interval"
+                    value={this.props.batch.properties.interval}
+                    onChange={(event, value) => this.props.onUpdateBatchIntervalValue(value)}
+                />
+            </div>
+        );
+    }
+
+    renderIntervalTypeDropdown() {
```

Comment:
The entire definition of renderIntervalTypeDropdown is repeated/duplicated 5 times in your latest changes, why not just introduce one function called renderUnitTypeDropdown(type, onUpdateType)? i.e. you can call this method passing in different 'type' value parameter and 'onUpdateType' function parameter. There is no need to duplicate the code 5 times.

Reply:
done. created the help methods and used them.